### ISSUE TITLE
Perform suitable static analysis of the jobshop model

### ISSUE THREAD

**Issue by ushnisha** [ushnisha/jobshop-minimal#23]
Static analysis can be used by the solver to speed it up or to reduce the search space or to come up with smarter algorithms (directional search) etc.  This could include:

(1) Determine the "level" of each task and workcenter (so that we can use that to perform upstream vs. downstream sweeps of certain kinds)

(2) Determine the criticality of workcenters.  This will help us solve overload problems on the critical resources first.  The meaning of the word critical is still unclear.  The solution should also define what criticality means while coming up with ideas for determining criticality of resources.

(3) Determine EPST and LPST of tasks.  Since the quantity of the taskplans also determines it length, the EPST and LPST are really dynamic quantities; a taskplan may be split into two smaller pieces and simultaneously assigned to two alternate resources; that would change the EPST/LPST calculations!  So, any such information should be a guide to the solver; and we must always remember that it can lead to sub-optimal plans (as with any heuristic planning approach).

(4) Determining if the jobshop can be broken up into independent subproblems that can be solved in parallel.

(5) Anything else?

**Comment by ushnisha** [comment]
Points (2) and (3) are not within the realm of static analysis.  They are dynamic in nature and change depending on the demand quantities, dates, workcenter availability information etc.  It is best to move these to a new issue and calculate or update these quantities during planning.

**Comment by ushnisha** [comment]
Fixed with commit: https://github.com/ushnisha/jobshop-minimal/commit/bfe7025fa032f811e7f089cbe56191de128fe314
